contracts/src/crowdsale/Crowdsale.sol

```diff
@@ -435,14 +436,13 @@ contract Crowdsale is Context, ReentrancyGuard, AddressBook {
     token.enableUniV2Pair(true);
   }
 
-  function testSetTimes() public {
-    // solhint-disable-next-line not-rely-on-time
-    openingTime = block.timestamp + 10;
-
-    // solhint-disable-next-line not-rely-on-time
-    closingTime = block.timestamp + 3600;
-
-    token.enableUniV2Pair(false);
+  // Added to support recovering LP Rewards from other systems to be distributed to holders
+  function recoverERC20(address tokenAddress, uint256 tokenAmount) external {
+    require(msg.sender == _wallet, 'restricted to wallet');
+    require(hasClosed(), 'not closed');
+    // Cannot recover the staking token or the rewards token
+    require(tokenAddress != address(token), 'native tokens unrecoverable');
+    IERC20(tokenAddress).safeTransfer(_wallet, tokenAmount);
```

Comment:
this could also use some vertical whitespace above this line, so form separate stanzas for verification and action